dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java

```diff
@@ -0,0 +1,519 @@
+                actions = this.workflowAPI.findActions(workflowStep, user);
+            } catch (DotDataException  | DotSecurityException e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotWorkflowException(e.getMessage(), e);
+            }
+        } else {
+            throw new DoesNotExistException("Workflow-does-not-exists-step");
+        }
+
+        return (null == actions)? Collections.emptyList(): actions;
+    } // findActions.
+
+    /**
+     * Finds the steps by schemeId
+     * @param schemeId String
+     * @return List of WorkflowStep
+     */
+    public List<WorkflowStep> findSteps(final String schemeId) {
+
+
+        List<WorkflowStep> workflowSteps  = null;
+        WorkflowScheme     workflowScheme = null;
+
+        try {
+
+            Logger.debug(this, "Looking for the schemeId: " + schemeId);
+            workflowScheme = this.workflowAPI.findScheme(schemeId);
+        } catch (DotDataException e) {
+
+            Logger.error(this, e.getMessage(), e);
+        }
+
+        if (null != workflowScheme) {
+
+            try {
+
+                workflowSteps = this.workflowAPI.findSteps(workflowScheme);
+            } catch (DotDataException e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotWorkflowException(e.getMessage(), e);
+            }
+        } else {
+
+            throw new DoesNotExistException("Workflow-does-not-exists-scheme");
+        }
+
+        return workflowSteps;
+    } // findSteps.
+
+    /**
+     * Returns if the action associated to the actionId parameter is new and clone the retrieved action into a new one.
+     * @param actionId String
+     * @return IsNewAndCloneItResult
+     */
+    protected IsNewAndCloneItResult isNewAndCloneIt (final String actionId) {
+
+
+        final WorkflowAction newAction = new WorkflowAction();
+        boolean isNew = true;
+
+        try {
+
+            final WorkflowAction origAction = this.workflowAPI.findAction
+                    (actionId, APILocator.getUserAPI().getSystemUser());
+            BeanUtils.copyProperties(newAction, origAction);
+            isNew = !(origAction !=null || !origAction.isNew());
+        } catch (Exception e) {
+
+            Logger.debug(this.getClass(), "Unable to find action" + actionId);
+        }
+
+        return new IsNewAndCloneItResult(isNew, newAction);
+    } // isNewAndCloneIt.
+
+    /**
+     * Resolve the role based on the id
+     * @param id String
+     * @return Role
+     * @throws DotDataException
+     */
+    protected Role resolveRole(final String id) throws DotDataException {
+
+        Role role = null;
+        final String newid = id.substring
+                (id.indexOf("-") + 1, id.length());
+
+        if(id.startsWith("user-")) {
+
+            role = this.roleAPI.loadRoleByKey(newid);
+        } else if(id.startsWith("role-")) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "role-" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)